olive/passes/pytorch/gptq.py

```diff
@@ -166,20 +167,35 @@ def _run_for_config(
         def get_onnx_quant_linear(*args, **kwargs):
             return QuantLinear
 
-        if hasattr(pytorch_model, "config") and pytorch_model.config.model_type in GPTQ_CAUSAL_LM_MODEL_MAP:
-            model_type = pytorch_model.config.model_type
-            model_class = GPTQ_CAUSAL_LM_MODEL_MAP[model_type]
-            quantized_model = model_class(pytorch_model, False, quantize_config)
-        else:
-            quantized_model = BaseGPTQForCausalLM(pytorch_model, False, quantize_config)
-            if not (config["layers_block_name"] and config["outside_layer_modules"] and config["inside_layer_modules"]):
+        model_type = pytorch_model.config.model_type if hasattr(pytorch_model, "config") else ""
+        model_class = GPTQ_CAUSAL_LM_MODEL_MAP.get(model_type, BaseGPTQForCausalLM)
+        quantized_model = model_class(pytorch_model, False, quantize_config)
+
+        if config["outside_layer_modules"]:
+            quantized_model.outside_layer_modules = config["outside_layer_modules"]
+        elif model_type not in GPTQ_CAUSAL_LM_MODEL_MAP:
+            if model_type in MODEL_OUTSIDE_LAYER_MODULES:
+                quantized_model.outside_layer_modules = MODEL_OUTSIDE_LAYER_MODULES[model_type]
+            else:
                 raise ValueError(
-                    "Can't get layers_block_name to quantize automatically, "
-                    "please set layers_block_name, outside_layer_modules and inside_layer_modules in config."
+                    "Can't get outside_layer_modules to quantize automatically, please provide it in config."
                 )
-            quantized_model.layers_block_name = config["layers_block_name"]
-            quantized_model.outside_layer_modules = config["outside_layer_modules"]
+
+        if config["inside_layer_modules"]:
             quantized_model.inside_layer_modules = config["inside_layer_modules"]
+        elif model_type not in GPTQ_CAUSAL_LM_MODEL_MAP:
+            if model_type in MODEL_INSIDE_LAYER_MODULES:
+                quantized_model.inside_layer_modules = MODEL_INSIDE_LAYER_MODULES[model_type]
+            else:
+                raise ValueError(
+                    "Can't get inside_layer_modules to quantize automatically, please provide it in config."
+                )
+
+        if model_type not in GPTQ_CAUSAL_LM_MODEL_MAP:
```

Comment:
should we also do the same for `layer_block_name` and remove the default value for this? If you look at the autogptq base, there is nothing special about layer_block_name compared to the other two https://github.com/AutoGPTQ/AutoGPTQ/blob/097dd04e2fd02bc6664908cd91b7ef8c6302a332/auto_gptq/modeling/_base.py#L88

Reply:
also since the logic is the same, could you abstract the logic and parameterize the three?